catalog/spatial/wfs/spatial-wfs-featuretransformer/src/main/java/org/codice/ddf/spatial/ogc/wfs/featuretransformer/impl/WfsTransformerProcessor.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.wfs.featuretransformer.impl;
+
+import ddf.catalog.data.Metacard;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Optional;
+import org.codice.ddf.spatial.ogc.wfs.featuretransformer.FeatureTransformer;
+import org.codice.ddf.spatial.ogc.wfs.featuretransformer.WfsMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class WfsTransformerProcessor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WfsTransformerProcessor.class);
+  private List<FeatureTransformer> transformerServiceList;
+
+  public WfsTransformerProcessor(List<FeatureTransformer> transformerServiceList) {
+    this.transformerServiceList = transformerServiceList;
+  }
+
+  public Optional<Metacard> apply(String featureMember, WfsMetadata metadata) {
+
+    for (FeatureTransformer featureTransformer : transformerServiceList) {
+
+      try (InputStream featureMemberInputStream =
+          new BufferedInputStream(new ByteArrayInputStream(featureMember.getBytes()))) {
```

Comment:
I'm adding a null check for `featureMember`.

The try-with-resources is inside the loop because we need a fresh InputStream for each call to `featureTransformer.apply()`. Although `ByteArrayInputStream` will never *actually* throw `IOException`, `BufferedInputStream`'s contract makes this necessary. If `IOException` were thrown by `BufferedInputStream` - for whatever reason - then I don't think we necessarily want to stop processing altogether. I think we only want to proceed on to the next `FeatureTransfromer` with a new `InputStream`.